### ISSUE TITLE
mod_sql_mysql needs to quote table names due to reserved MySQL keywords

### ISSUE THREAD

**Issue by ubawurinna** [proftpd/proftpd#1212]
mod_sql configuration sample and the default tablename for SQLGroupInfo use a MySQL reserved keyword
=> SQLGroupInfo **groups** groupname gid members

Since MySQL 8.02 **groups** is a reserved keyword, and you don't use quoting in your queries.
=>
SELECT * FROM groups; => fail since MySQL 8.0.2
SELECT * FROM \`groups\`; => with quoting, it's ok

Affected files:
_doc\Configuration.html
doc\contrib\mod_sql.html
sample-configurations\mod_sql.conf_

**Comment by Castaglia** [comment]
Changing the defaults for `mod_sql` would break existing implementations for non-MySQL 8.x using sites.  So that is not something to be done lightly.

What are the errors/failures you see, when using the default configuration and MySQL 8.x?

**Comment by ubawurinna** [comment]
`mod_sql/4.3: unrecoverable backend error: (1064) You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'groups WHERE (gid = 5500) LIMIT 1' at line 1`

`query "SELECT groupname FROM groups WHERE (gid = 5500) LIMIT 1"
entering     mysql cmd_close
connection 'default' count is now 1
exiting      mysql cmd_close
exiting      mysql cmd_select
unrecoverable backend error
error: '1064'
message: 'You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'groups WHERE (gid = 5500) LIMIT 1' at line 1'`

**Comment by ubawurinna** [comment]
You don't need to change defaults for `mod_sql` but you should use backticks for table and column identifiers in your query builder for mysql.

For example the query:
`SELECT groupname FROM groups WHERE (gid = 5500) LIMIT 1;`
would become:
```SELECT `groupname` FROM `groups` WHERE (`gid` = 5500) LIMIT 1;```

### Here are the errors I get in the logs
**Error in proftd.log**

`mod_sql/4.3: unrecoverable backend error: (1064) You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'groups WHERE (gid = 5500) LIMIT 1' at line 1`

**Error in sql.log**

```
query "SELECT groupname FROM groups WHERE (gid = 5500) LIMIT 1"
entering     mysql cmd_close
connection 'default' count is now 1
exiting      mysql cmd_close
exiting      mysql cmd_select
unrecoverable backend error
error: '1064'
message: 'You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'groups WHERE (gid = 5500) LIMIT 1' at line 1'
```

a

**Comment by Castaglia** [comment]
Hrm.  I can reproduce this behavior locally.

What makes it annoying is that the MySQL-provided [`mysql_real_escape_string`](https://dev.mysql.com/doc/c-api/8.0/en/mysql-real-escape-string.html) does not properly handle the quoting of such reserved keywords, nor does the newer [`mysql_real_escape_string_quote`](https://dev.mysql.com/doc/c-api/8.0/en/mysql-real-escape-string-quote.html) function.  This means the application (`mod_sql`, or more properly `mod_sql_mysql`) has to do it.

This particular reserved keyword behavior for the default "groups" table name is specific to MySQL 8.x.  It does not apply to Postgres, SQLite, or the other databases supposed; nor does it apply to older MySQL versions.  The `mod_sql` module is database-agnostic, thus trying to use backticks there is at the wrong abstraction layer -- we do *not* want MySQL-specific tweaks/workarounds in the `mod_sql` module.  Thus the `mod_sql_mysql` module will need specific code to deal with this.

**Comment by Castaglia** [comment]
This should now be fixed in master; the fix has been backported to the 1.3.7 branch.  Thanks!
